ena-submission/scripts/create_assembly.py

```diff
@@ -0,0 +1,527 @@
+
+import click
+import pytz
+import yaml
+from ena_submission_helper import (
+    CreationResults,
+    check_ena,
+    create_chromosome_list,
+    create_ena_assembly,
+    create_fasta,
+    create_manifest,
+    get_ena_config,
+)
+from ena_types import (
+    AssemblyChromosomeListFile,
+    AssemblyChromosomeListFileObject,
+    AssemblyManifest,
+    AssemblyType,
+    ChromosomeType,
+    MoleculeType,
+)
+from notifications import SlackConfig, send_slack_notification, slack_conn_init
+from submission_db_helper import (
+    AssemblyTableEntry,
+    Status,
+    StatusAll,
+    add_to_assembly_table,
+    db_init,
+    find_conditions_in_db,
+    find_errors_in_db,
+    find_waiting_in_db,
+    update_db_where_conditions,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    encoding="utf-8",
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)8s (%(filename)20s:%(lineno)4d) - %(message)s ",
+    datefmt="%H:%M:%S",
+)
+
+
+@dataclass
+class Config:
+    organisms: list[dict[str, str]]
+    backend_url: str
+    keycloak_token_url: str
+    keycloak_client_id: str
+    username: str
+    password: str
+    db_username: str
+    db_password: str
+    db_host: str
+    db_name: str
+    unique_project_suffix: str
+    ena_submission_url: str
+    ena_submission_password: str
+    ena_submission_username: str
+    ena_reports_service_url: str
+    slack_hook: str
+    slack_token: str
+    slack_channel_id: str
+
+
+def create_chromosome_list_object(
+    unaligned_sequences: dict[str, str], seq_key: dict[str, str]
+) -> str:
+    # Use https://www.ebi.ac.uk/ena/browser/view/GCA_900094155.1?show=chromosomes as a template
+    # Use https://www.ebi.ac.uk/ena/browser/view/GCA_000854165.1?show=chromosomes for multi-segment
+
+    chromosome_type = ChromosomeType.SEGMENTED
+
+    entries: list[AssemblyChromosomeListFileObject] = []
+
+    if len(unaligned_sequences.keys()) > 1:
+        for segment_name, item in unaligned_sequences.items():
+            if item:  # Only list sequenced segments
+                entry = AssemblyChromosomeListFileObject(
+                    object_name=f"{seq_key["accession"]}.{seq_key["version"]}_{segment_name}",
+                    chromosome_name=segment_name,
+                    chromosome_type=chromosome_type,
+                )
+                entries.append(entry)
+    else:
+        entry = AssemblyChromosomeListFileObject(
+            object_name=f"{seq_key["accession"]}.{seq_key["version"]}",
+            chromosome_name="main",
+            chromosome_type=chromosome_type,
+        )
+        entries.append(entry)
+
+    return AssemblyChromosomeListFile(chromosomes=entries)
+
+
+def create_manifest_object(
+    config: Config,
+    sample_table_entry: dict[str, str],
+    project_table_entry: dict[str, str],
+    submission_table_entry: dict[str, str],
+    seq_key: dict[str, str],
+    group_key: dict[str, str],
+    test=False,
+) -> str:
```

Comment:
I think the result type is wrong, this function is returning a `AssemblyManifest`, isn't it?